### PR TITLE
Fix expression validation for variables with brackets

### PR DESCRIPTION
**Pull request by D8H**

### CODE REVIEW

File: Core/tests/ExpressionParser2.cpp [hunk 1]
```diff
@@ -1519,6 +1519,14 @@ TEST_CASE("ExpressionParser2", "[common][events]") {
       node->Visit(validator);
       REQUIRE(validator.GetFatalErrors().size() == 0);
     }
+    {
```
Comment (4ian) [inline Core/tests/ExpressionParser2.cpp:1522]: This should probably be in a different test "Variable with bracket accessor and operators".

Add checks for *, -, /

If you can also make the test to check what is code generated, that would be safer.

Comment (4ian) [reply Core/tests/ExpressionParser2.cpp:1522]: Also add tests for "Variable with nested bracket accessor and operator" which does something like:

`MySceneStructureVariable[MySceneStructureVariable2["MyChild"]] * 2` (with all operators) 
`MySceneStructureVariable[MySceneStructureVariable2["MyChild"] * 2] * 3` (with all operators)

both for the validator and for code generation.

File: Core/GDCore/IDE/Events/ExpressionValidator.h [hunk 2]
```diff
@@ -293,12 +293,14 @@ class GD_CORE_API ExpressionValidator : public ExpressionParser2NodeWorker {
     forbidsUsageOfBracketsBecauseParentIsObject = false;
 
     Type currentParentType = parentType;
+    Type currentChildType = childType;
```
Comment (4ian) [inline Core/GDCore/IDE/Events/ExpressionValidator.h:296]: Should you also save and restore `forbidsUsageOfBracketsBecauseParentIsObject`?

Comment (4ian) [reply Core/GDCore/IDE/Events/ExpressionValidator.h:296]: Well, it's actually reseted just before. Is there a risk the `Visit` is setting it back to `true` (meaning we should again reset it to false just after)?

Comment (D8H) [reply Core/GDCore/IDE/Events/ExpressionValidator.h:296]: I think it won't be a problem. It's only a message between a `VariableNode` and its `VariableBracketAccessorNode` if any.

**Pull request merged**